# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/mnemoc/uclibc/patches/util-linux-30-hotfix.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- util-linux-2.11z/mount/fstab.c.orig	2003-12-03 15:28:22.000000000 -0700
+++ util-linux-2.11z/mount/fstab.c	2003-12-03 15:28:41.000000000 -0700
@@ -342,7 +342,7 @@
 /* Ensure that the lock is released if we are interrupted.  */
 static void
 handler (int sig) {
-     die (EX_USER, "%s", sys_siglist[sig]);
+     die (EX_USER, "%s", strsignal(sig));
 }
 
 static void
--- /dev/null	2003-09-07 01:55:59.000000000 -0600
+++ util-linux-2.11z/mount/swapargs.h	2003-12-03 15:44:50.000000000 -0700
@@ -0,0 +1,3 @@
+#define SWAPON_HAS_TWO_ARGS
+#include <asm/page.h>
+#include <sys/swap.h>
--- util-linux-2.11z/mount/swap.configure.orig	2003-12-03 15:43:24.000000000 -0700
+++ util-linux-2.11z/mount/swap.configure	2003-12-03 15:45:33.000000000 -0700
@@ -1,6 +1,8 @@
 # Find out whether we can include <sys/swap.h>
 # and whether libc thinks that swapon() has two arguments.
 
+exit 0
+
 # Prepare test
 CC=${CC-cc}
 compile="$CC -o conftest conftest.c >/dev/null 2>&1"
--- ./partx/solaris.c.orig	2000-03-21 18:08:51.000000000 -0400
+++ ./partx/solaris.c	2004-02-11 23:40:25.000000000 -0300
@@ -5,8 +5,6 @@
 #define SOLARIS_X86_NUMSLICE	8
 #define SOLARIS_X86_VTOC_SANE	(0x600DDEEEUL)
 
-typedef int daddr_t;		/* or long - check */
-
 struct solaris_x86_slice {
 	unsigned short	s_tag;		/* ID tag of partition */
 	unsigned short	s_flag;		/* permision flags */
